hoomd/hpmc/pytest/test_compute_free_volume.py

```diff
@@ -0,0 +1,64 @@
+import hoomd
+import pytest
+import numpy as np
+
+
+def test_before_attaching():
+    mc = hoomd.hpmc.integrate.Sphere()
+    mc.shape["A"] = {'diameter': 1.0}
+    mc.shape["B"] = {'diameter': 0.2}
+    mc.depletant_fugacity["B"] = 1.5
+    free_volume = hoomd.hpmc.compute.FreeVolume(mc, test_type='B', nsample=100)
+
+    assert free_volume.mc == mc
+    assert free_volume.test_particle_type == 'B'
+    assert free_volume.num_samples == 100
+    assert free_volume.free_volume is None
+
+
+def test_after_attaching(simulation_factory, lattice_snapshot_factory):
+    snap = lattice_snapshot_factory(particle_types=['A', 'B'])
+    sim = simulation_factory(snap)
+    mc = hoomd.hpmc.integrate.Sphere()
+    mc.shape["A"] = {'diameter': 1.0}
+    mc.shape["B"] = {'diameter': 0.2}
+    mc.depletant_fugacity["B"] = 1.5
+    sim.operations.add(mc)
+
+    sim.operations._schedule()
+    free_volume = hoomd.hpmc.compute.FreeVolume(mc, test_type='B', nsample=100)
+
+    sim.operations.add(free_volume)
+    assert len(sim.operations.computes) == 1
+    sim.run(0)
+
+    assert free_volume.test_particle_type == 1
+    assert free_volume.num_samples == 100
+
+    sim.run(10)
+    assert isinstance(free_volume.free_volume, float)
+
+_radii = [(0.25, 0.05),
+          (0.4, 0.05),
+          (0.7, 0.17)]
+
+@pytest.mark.parametrize("radius1, radius2", _radii)
+def test_validation_systems(simulation_factory,
+                            two_particle_snapshot_factory,
+                            lattice_snapshot_factory,
+                            radius1,
+                            radius2):
+    free_volume = (7**3) * (1 - (4 / 3) * np.pi * (radius1 + radius2)**3)
```

Comment:
Nice test!

I'm not sure why it is failing in some configurations. I am going to investigate.